src/components/ha-md-dialog.ts

```diff
@@ -0,0 +1,132 @@
+import { MdDialog } from "@material/web/dialog/dialog";
+import { css } from "lit";
+import { customElement, property } from "lit/decorators";
+
+/**
+ * Based on the home assistant design: https://design.home-assistant.io/#components/ha-dialogs
+ *
+ */
+@customElement("ha-md-dialog")
+export class HaMdDialog extends MdDialog {
+  /**
+   * When true the dialog will not close when the user presses the esc key or press out of the dialog.
+   */
+  @property({ attribute: "disable-cancel-action", type: Boolean })
+  public disableCancelAction = false;
+
+  constructor() {
+    super();
+    if (typeof HTMLDialogElement === "function") {
+      this.open = true;
+    }
+  }
+
+  async connectedCallback() {
+    super.connectedCallback();
+    this.addEventListener("cancel", this._handleCancel);
+
+    // polyfill dialog for older browsers
+    if (typeof HTMLDialogElement !== "function" && this.shadowRoot?.host) {
```

Comment:
We probably should not do this in `connectedCallback` as it could happen more than once, and we only need to do this once?

Also, why do we check for `this.shadowRoot?.host`?